### PR TITLE
OCPBUGS-44244: use node node in deleted machine phases

### PR DESCRIPTION
**Pull request by kannon92**
This PR makes sure to add a node name to the DeletedMachine monitor.

Once that is added, we also added some unit tests to this code to make sure that we are correctly detecting UnexpectedNodeNotReady / UnexpectedUnreachable.

The main edge case we had to work around was when Machines get deleted and that node goes not ready. This is correct behavior but our test was flagging this as this code path does not go through a Machine Config change.

### CODE REVIEW

**Comment by kannon92** [comment]
/hold

I need to write some unit tests.

File: pkg/monitortests/machines/watchmachines/machines.go [hunk 1]
```diff
@@ -83,15 +83,24 @@ func startMachineMonitoring(ctx context.Context, m monitorapi.RecorderWriter, cl
...
+
+			nodeName := "<unknown>"
+			oldHasNodeRef := oldMachine != nil && oldMachine.Status.NodeRef != nil
+			newHasNodeRef := machine.Status.NodeRef != nil
+			if oldHasNodeRef {
+				nodeName = oldMachine.Status.NodeRef.Name
+			}
+
+			if newHasNodeRef {
+				nodeName = machine.Status.NodeRef.Name
```
Comment (dgoodwin) [inline pkg/monitortests/machines/watchmachines/machines.go:95]: So getting a correct node name into the interval annotation will allow your test logic to correctly correlate this deleting machine with a node not ready, and ignore it going forward? i.e. this will cause LESS failures without having to make the test flake.

Suggest explaining some of that in the commit message + PR when you finish this up and have added the unit tests.

Comment (kannon92) [reply pkg/monitortests/machines/watchmachines/machines.go:97]: I don't have a lot of real estate to work with for commit message. I added this to the description.

**Comment by kannon92** [comment]
cc @rphillips @deads2k @dgoodwin 

This is ready. I want to run the tests off of this PR so please don't merge until we confirm this works.

**Comment by rphillips** [comment]
Nice. This looks good... Let's see how the tests go.

**Comment by kannon92** [comment]
/hold cancel

Something is up with these tests..

**Comment by kannon92** [comment]
/retest

/hold cancel

Confirmed that the machines have the right node names in the MachinePhase Intervals.

**Comment by kannon92** [comment]
@deads2k @dgoodwin @rphillips 
This PR is ready to merge. PTAL.

File: pkg/monitortests/node/watchnodes/node_test.go [hunk 2]
```diff
@@ -55,3 +57,283 @@ func TestNodeRoles(t *testing.T) {
...
+					},
+					From: utility.SystemdJournalLogTime("Nov 11 19:46:00"),
+					To:   utility.SystemdJournalLogTime("Nov 11 19:46:00"),
+				},
+			},
+			expected:         []string{"node/node1 - reason/UnexpectedNotReady unexpected node not ready at from: 2024-11-11 19:46:00 +0000 UTC - to: 2024-11-11 19:46:00 +0000 UTC"},
+			unexpectedReason: monitorapi.NodeUnexpectedReadyReason,
+		},
+		{
+			// OCPBUG-44244: if machine monitor does not have node name than we will get no failures
```
Comment (dgoodwin) [inline pkg/monitortests/node/watchnodes/node_test.go:96]: This test looks to expect a failure. Just a comment that needs updating?

File: pkg/monitortests/node/watchnodes/node_test.go [hunk 3]
```diff
@@ -55,3 +57,283 @@ func TestNodeRoles(t *testing.T) {
...
+					},
+					From: utility.SystemdJournalLogTime("Nov 11 19:46:00"),
+					To:   utility.SystemdJournalLogTime("Nov 11 19:46:00"),
+				},
+			},
+			expected:         []string{"node/node1 - reason/UnexpectedUnreachable unexpected node unreachable at from: 2024-11-11 19:46:00 +0000 UTC - to: 2024-11-11 19:46:00 +0000 UTC"},
+			unexpectedReason: monitorapi.NodeUnexpectedUnreachableReason,
+		},
+		{
+			// OCPBUG-44244: if machine monitor does not have node name than we will get no failures
```
Comment (dgoodwin) [inline pkg/monitortests/node/watchnodes/node_test.go:224]: expects a failure as well

**Review by dgoodwin — COMMENTED** [review]
Looks good to me just a couple nits.

File: pkg/monitortests/node/watchnodes/node_test.go (hunk 2 above)
Comment (kannon92) [reply pkg/monitortests/node/watchnodes/node_test.go]: yea. updated.

File: pkg/monitortests/node/watchnodes/node_test.go (hunk 3 above)
Comment (kannon92) [reply pkg/monitortests/node/watchnodes/node_test.go]: updated.
